msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/HelperAndUtilityTests.java

```diff
@@ -0,0 +1,72 @@
+package com.microsoft.aad.msal4j;
+
+import org.junit.jupiter.api.Test;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateEncodingException;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+class HelperAndUtilityTests {
+
+    @Test
+    void JwtHelper_buildJwt_ValidSha1AndSha256Assertions() throws MsalClientException, CertificateEncodingException, NoSuchAlgorithmException {
+        ClientCertificate clientCertificateMock = mock(ClientCertificate.class);
+        when(clientCertificateMock.privateKey()).thenReturn(TestHelper.getPrivateKey());
+        when(clientCertificateMock.publicCertificateHash()).thenReturn("certificateHash");
+        when(clientCertificateMock.publicCertificateHash256()).thenReturn("certificateHash256");
+        when(clientCertificateMock.getEncodedPublicKeyCertificateChain()).thenReturn(Arrays.asList("cert1", "cert2"));
+
+        String clientId = "clientId";
+        String audience = "https://login.microsoftonline.com/common/oauth2/v2.0/token";
+
+        //Sha256 assertion
+        ClientAssertion clientAssertion = JwtHelper.buildJwt(clientId, clientCertificateMock, audience, true, false);
+
+        assertNotNull(clientAssertion);
+        assertNotNull(clientAssertion.assertion());
+
+        // Verify JWT structure (header.payload.signature)
+        String jwt = clientAssertion.assertion();
+        String[] jwtParts = jwt.split("\\.");
+        assertEquals(3, jwtParts.length, "JWT should have three parts");
+
+        // Decode and verify headers
+        String headerJson = new String(Base64.getUrlDecoder().decode(jwtParts[0]));
+        assertTrue(headerJson.contains("\"alg\":\"RS256\""), "Header should specify RS256 algorithm");
```

Comment:
It should be PS256 . RS256 means PKCS1 padding, which is not approved.

Reply:
These comments have more context: https://github.com/AzureAD/microsoft-authentication-library-for-java/pull/941#discussion_r2060023717